deeplake/core/index_maintenance.py

```diff
@@ -134,7 +134,9 @@ def index_operation_type_vectorstore(
         if not below_threshold:
             return INDEX_OP_TYPE.CREATE_INDEX
     else:
-        if (
```

Comment:
Which case does this correspond to? Was it previously erroring out? If yes can we add test for this edge case? I do get that it the case when there are no samples in the dataset, am I right @sounakr?

Reply:
Initially due to some pre-exiting bug vectorstore saw some scenario where empty dataset is coming till index maintenance. This is a error case and finally it will fail down the line. Just to be safe placed a check so that index maintenance doesn't take any action for those cases and become NO OP.

Reply:
Yes this is edge case.